cmd/update-rules/main.go

```diff
@@ -142,7 +146,9 @@ func UpdateRules(rules *config.RepositoryRules, branch, goVer string) {
 		}
 
 		// update the rules for branch and its dependencies
-		updateBranchRules(&newBranchRule, branch, goVer)
+		if !removeRules {
```

Comment:
If we check for `removeRules` only here, it will result in existing branch rule to be replaced with `master` instead of removal of the rule.

eg: if `release-1.25` branch needs to be removed, this will result in 2 entries for master. One being the original one and the other being `release-1.25` overwritten as master.